app/api/routes/image_queries.py

```diff
@@ -22,32 +20,17 @@
 router = APIRouter()
 
 
-async def validate_image(request: Request) -> Image.Image:
-    """
-    Validate the image file contained in the request body and return a PIL Image object.
-    :param file: The uploaded image file to be validated.
-    :return: A PIL Image object if the file is a valid image.
-    :raises HTTPException: If the file is not an image or cannot be processed.
-    """
+async def validate_content_type(request: Request) -> str:
     if not request.headers.get("Content-Type", "").startswith("image/"):
         raise HTTPException(
             status_code=status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, detail="Request body must be image bytes"
         )
+    return request.headers.get("Content-Type", "")
 
-    image_bytes = await request.body()
-    try:
-        # Attempt to open the image
-        image = Image.open(BytesIO(image_bytes))
 
-        # Image.open() does not fully process the image data. It's possible for Image.open()
-        # to succeed but then fail when the image data is actually being processed.
-        # To ensure that the image can be fully processed, we call img.load() to force loading
-        # the entire image. If this fails, we know that the image is invalid.
-        image.load()
-    except IOError as ex:
-        logger.error("Failed to load image", exc_info=True)
-        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Invalid input image") from ex
-    return image
+async def validate_image_bytes(request: Request, content_type: str = Depends(validate_content_type)) -> bytes:
```

Comment:
note: Dependency injection in fastapi forms a DAG. Nodes of the DAG do not need to be recomputed. So validate_content_type is computed once per request and the result is used in multiple places. In this case, it is to perform validation before attempting to return the image. Below, it is actually to get the content-type.